lib/deep-equal.js

```diff
@@ -200,10 +200,6 @@ function deepEqualCyclic(actual, expectation, match) {
             !isArrayType(expectation) &&
             !isArguments(expectation);
         if (isActualNonArrayIterable || isExpectationNonArrayIterable) {
-            if (!isActualNonArrayIterable || !isExpectationNonArrayIterable) {
-                return false;
-            }
-
```

Comment:
This was, as far as I can tell, unreachable code. If one of these isn't a non array iterable but the other is, earlier branches in the function catch that scenario.